source/boundary_temperature/interface.cc

```diff
@@ -215,39 +162,93 @@ namespace aspect
     {
-      return boundary_temperature_objects;
+      std_cxx11::get<dim>(registered_plugins).register_plugin (name,
+                                                               description,
+                                                               declare_parameters_function,
+                                                               factory_function);
     }
 
 
+
     template <int dim>
     void
     Manager<dim>::declare_parameters (ParameterHandler &prm)
     {
+      prm.enter_subsection ("Model settings");
+      {
+        prm.declare_entry ("Fixed temperature boundary indicators", "",
+                           Patterns::List (Patterns::Anything()),
+                           "A comma separated list of names denoting those boundaries "
+                           "on which the temperature is fixed and described by the "
+                           "boundary temperature objects selected in the "
+                           "<Boundary temperature model/List of model names> parameter "
+                           "of this input file. All boundary indicators used by the geometry "
+                           "but not explicitly listed here will end up with no-flux "
+                           "(insulating) boundary conditions."
+                           "\n\n"
+                           "The names of the boundaries listed here can either by "
+                           "numbers (in which case they correspond to the numerical "
+                           "boundary indicators assigned by the geometry object), or they "
+                           "can correspond to any of the symbolic names the geometry object "
+                           "may have provided for each part of the boundary. You may want "
+                           "to compare this with the documentation of the geometry model you "
+                           "use in your model."
+                           "\n\n"
+                           "Each entry in this list has to be either a single boundary "
+                           "identifier as described above, or a pattern of the type "
+                           "'boundary identifier : plugin name', where plugin name is "
+                           "one of the names given in the "
+                           "<Boundary temperature model/List of model names> parameter. "
+                           "If only a boundary identifier is given, the boundary temperature "
+                           "of all active boundary temperature plugins will be combined to "
+                           "compute the boundary temperature. If only one plugin is given, "
+                           "only this plugin is used. The same boundary identifier can "
+                           "appear multiple times with different plugin names, in which "
+                           "case all given plugins will be combined to compute the boundary "
```

Comment:
After thinking about this for a while, I guess the most reasonable approach I see would be to change this format to allow something like:

set Fixed temperature boundary indicators   = top : box & function & constant, left : box & initial temperature, bottom

If there is no colon, all plugins are applied (as it is right now, so it's backwards compatible). I guess that means duplicating the list of plugins in the 'List of model names' parameter, but I don't really see any other way to do this, and it would only come up in complex models. 

So I guess the only other question we have to think about here is: Do we ever want to allow having the same kind of plugin, but with different parameters, for different boundaries (like 2 different functions)? Because I don't really see how we would put that into this framework, so we would essentially say that this is someting we do not want to change in the near future.

Reply:
I haven't looked at this since my last review, but just about the format: instead of `&` we've generally used `;` as a top-level separator between entries I believe.